outbox/src/main/kotlin/com/ritense/outbox/OutboxMessageRepository.kt

```diff
@@ -31,7 +31,7 @@ interface OutboxMessageRepository : JpaRepository<OutboxMessage, UUID> {
      */
     @QueryHints(value = [QueryHint(name = JPA_LOCK_TIMEOUT, value = SKIP_LOCKED)])
     @Lock(PESSIMISTIC_WRITE)
-    fun findTopByOrderByCreatedOnAsc(): OutboxMessage?
+    fun findFirstBy(): OutboxMessage?
```

Comment:
On Mysql this query: 
`SELECT * from outbox_message ORDER BY created_on ASC LIMIT 1 FOR UPDATE SKIP LOCKED`
Will lock almost all rows in the table. Not sure why. I think its due to wrongly 'gap-locking' some rows.

On Postgres 'gap-locking' also happens but is done correctly